Add email functionality with Nodemailer for contact form submissions

adamski83 opened the pull request.

adamski83 committed:
Add email functionality with Nodemailer for contact form submissions

adamski83 committed:
Revert "Implement contact form with validation and submission handling"

This reverts commit 938847f554d96dcea05b59d72d905dd2de755554.

adamski83 committed:
Implement contact form with validation, email sending, and auto-reply functionality

adamski83 committed:
Refactor email handling by introducing email rendering functions and removing the EmailService class

fengerkamil commented on app/api/contact/route.ts:
```diff
@@ -0,0 +1,106 @@
+				phone: validatedData.phone,
+				subject: validatedData.subject,
+				message: validatedData.message,
+			}),
```
wystarczy `html: renderContactEmail(validatedData)`

fengerkamil commented on app/kontakt/page.tsx:
```diff
@@ -1,147 +1,234 @@
-import { useContactForm } from "@/hooks/useContactForm";
 import Image from "next/image";
+import { useState } from "react";
+import { toast } from "sonner";
```
dzięki za info, bardzo fajna libka

fengerkamil commented on app/kontakt/page.tsx:
```diff
@@ -1,147 +1,234 @@
+		} finally {
+			setIsSubmitting(false);
+		}
+	};
```
Znacznie lepszy, w połączeniu z zodem byłby react hook form niż pisanie formularza od zera
To samo tyczy się fetcha, axios to podstawowa kwestia każdego projektu, fetch jest spoko do eksperymentowania na szybko :D

fengerkamil commented on app/kontakt/page.tsx:
```diff
@@ -1,147 +1,234 @@
+									className="block text-sm font-medium text-white mb-2"
+								>
+									Imię *
+								</label>
```
To mógłby byc osobny komponent aby nie powielać klas w każdym labelu, co gdy przyjdzie potrzeba zmiany jego stylu? Będziesz zmieniał klasy w kilkunastu miejscach jeśli ilość formularzy wzrośnie? Label powinien mieć propsy `for` oraz `required`

fengerkamil commented on app/kontakt/page.tsx:
```diff
@@ -1,147 +1,234 @@
+									name="name"
+									value={formData.name}
+									onChange={handleChange}
+									className="w-full px-3 py-2 border border-gray-300 rounded-md focus:outline-none focus:ring-2 focus:ring-orange-500 text-gray-900"
```
Zduplikowane inputy i ich klasy, zamiast osobnych reużywalnych komponentów.

fengerkamil commented on app/kontakt/page.tsx:
```diff
@@ -1,147 +1,234 @@
+								className="w-full bg-orange-900 text-white py-3 px-6 rounded-md hover:bg-orange-800 focus:outline-none focus:ring-2 focus:ring-orange-500 font-medium disabled:opacity-50 disabled:cursor-not-allowed transition-colors"
 							>
-								{isSubmitting ? "WYSYŁANIE..." : "WYŚLIJ"}
+								{isSubmitting ? "WYSYŁANIE..." : "WYSŁAĆ"}
```
tekst z małej, ostylować go trzeba na uppercase odpowiednią klasą tailwind

fengerkamil commented on app/main/header.tsx:
```diff
@@ -9,12 +9,12 @@ export default function Header() {
+				<div className="flex flex-col sm:flex-row gap-2 ml-3.5 mr-3.5 mt-5 ">
+					<button className="btn btn-outline hover:bg-yellow-500">
 						<Phone className="mr-2" />
-						<span className=" sm:inline">+1234567890</span>
+						<span className="hidden sm:inline">+1234567890</span>
 					</button>
-					<button className="flex justify-around  btn btn-outline hover:bg-yellow-500">
+					<button className="btn btn-outline hover:bg-yellow-500">
 						<Mail className="mr-2" />
 						info@boki.com
 					</button>
```
To powinny być linki które pozwolą po kliknięciu wybrać ten numer telefonu od razu na telefonie, oraz otworzyć klienta pocztowego. Są do tego specjalne prefixy w hrefie znacznika `<a>`

fengerkamil commented on lib/email/emailRenderer.ts:
```diff
@@ -0,0 +1,97 @@
+		: "";
+
+	return `
+		<!DOCTYPE html>
```
Bardziej chodziło mi o coś takiego

https://react.dev/reference/react-dom/server/renderToString

adamski83 committed:
feat: update package.json scripts and dependencies, add ESLint and Prettier configurations

- Removed turbopack from dev script
- Updated linting scripts to include strict mode
- Added new dependencies: clsx, tailwind-merge, eslint-plugin-import, eslint-plugin-jsx-a11y, eslint-plugin-react, eslint-plugin-react-hooks, husky, and lint-staged
- Updated dev dependencies: @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, and eslint-config-next
- Added .eslintrc.json for ESLint configuration
- Added .prettierignore to exclude certain files from formatting
- Added VSCode extensions recommendations
- Implemented contact service for form submission
- Created reusable UI components: Checkbox, Input, Textarea
- Developed custom hook for handling contact form submission with Axios
- Set up Axios instance with interceptors for request and response logging
- Added utility function for class name merging
